internal/internal_task_handlers.go

```diff
@@ -260,6 +260,28 @@ func isDecisionEvent(eventType s.EventType) bool {
 	}
 }
 
+// isDecisionEventForReplay is different from isDecisionEvent because during replays
+// we want to intentionally ignore workflow complete/fail/cancel/continueasnew events so that
+// decision tree replays matches with the workflow processing respond tasks
+func isDecisionEventForReplay(eventType s.EventType) bool {
```

Comment:
I has tried something similar a while ago (a bit crude in comparison) but then dropped it because it was messing up the normal task processing and history replay. Are the integration tests passing for you?

Reply:
Tests are passing now.